### PR TITLE
Remove the need for a python env in CI

### PR DESCRIPTION
**Pull request by adrienbernede**
This pull requests updates BLT to use its new xml -> junit logic.

The need for a python environment with lxml is replace by the use of xsltproc and a conversion template provided by BLT.

CI simplified, local scripts removed. 

Thanks @white238

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
This is awesome!

**Comment by adrienbernede** [comment]
LGTM

File: scripts/gitlab/build_and_test.sh [hunk 1]
```diff
@@ -112,7 +112,7 @@ then
     $cmake_exe \
       -C ${hostconfig_path} \
       ${project_dir}
-    if ! $cmake_exe --build . -j; then
+    if ! $cmake_exe --build . -j 12 ; then
```
Comment (davidbeckingsale) [inline scripts/gitlab/build_and_test.sh:115]: Why the change to `-j 12`?

Comment (adrienbernede) [reply scripts/gitlab/build_and_test.sh:115]: I think @kab163 changed the number of jobs for RAJA (from `-j` to `-j32` or so). I thought I would do the same here. Although not related to this PR...

Comment (adrienbernede) [reply scripts/gitlab/build_and_test.sh:115]: I can revert that for sure.

Comment (adrienbernede) [reply scripts/gitlab/build_and_test.sh]: Done

**Comment by adrienbernede** [comment]
LGTM

**Comment by davidbeckingsale** [comment]
LGTM

**Review by davidbeckingsale — APPROVED** [review]
